src/inference/inference.ts

```diff
@@ -140,41 +170,31 @@ export class Inference {
           ' documents. See https://docs.pinecone.io/models for supported models.'
       );
     }
-    // Destructure `options` with defaults
-    // Note: If the user passes in key:value pairs in `options` that are not the following, they are ignored
+
     const {
       topN = documents.length,
       returnDocuments = true,
-      rankFields = ['text'],
       parameters = {},
+      rankFields = undefined,
     } = options;
 
-    // Allow documents to be passed a list of strings, or a list of objs w/at least a `text` key:
-    let newDocuments: Array<{ [key: string]: string }> = [];
-    if (typeof documents[0] === 'object' && !('text' in documents[0])) {
-      throw new PineconeArgumentError(
-        '`documents` can only be a list of strings or a list of objects with at least a `text` key, followed by a' +
-          ' string value'
-      );
-    } else if (typeof documents[0] === 'string') {
-      newDocuments = documents.map((doc) => {
-        return { text: doc as string };
-      });
-    } else {
-      newDocuments = documents as Array<{ [key: string]: string }>;
-    }
+    // Initialize computedRankFields with rankFields or default to 'text'
+    const computedRankFields = rankFields ?? ['text'];
+
+    // Validate and standardize documents to ensure they are in object format
+    const newDocuments = documents.map((doc) =>
+      typeof doc === 'string' ? { text: doc } : doc
+    );
 
-    // Ensure all rankFields, if passed, are present in each document
-    if (rankFields.length > 0) {
-      newDocuments.forEach((doc, index) => {
-        rankFields.forEach((field) => {
-          if (!(field in doc)) {
-            throw new PineconeArgumentError(
-              `The \`rankField\` value you passed ("${field}") is missing in the document at index ${index}`
-            );
-          }
-        });
-      });
+    if (!options.rankFields) {
+      // If each obj in newDocs does not have a 'text' field, throw an error
+      if (
+        !newDocuments.every((doc) => typeof doc === 'object' && 'text' in doc)
+      ) {
```

Comment:
The server will validate that the documents contain the fields they're supposed to, so it's not entirely necessary to validate on the client. However, if we want to validate on the client, why not also validate the case where a custom rank field was provided?

Reply:
oh yeah good call, I remove this part actually (the validation part)